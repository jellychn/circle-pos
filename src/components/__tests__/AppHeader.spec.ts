import { shallowMount } from '@vue/test-utils'
import { describe, it, expect, vi } from 'vitest'
import { useRouter } from 'vue-router'
import AppHeader from '../AppHeader.vue'

vi.mock('vue-router')
// @ts-ignore:next-line
vi.mocked(useRouter).mockReturnValue({
  push: vi.fn()
})

describe('AppHeader.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(AppHeader)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('calls navigateToHome when logo is clicked', async () => {
    const wrapper = shallowMount(AppHeader)

    await wrapper.find('.logo').trigger('click')

    expect(useRouter().push).toHaveBeenCalledWith('/')
  })
})
