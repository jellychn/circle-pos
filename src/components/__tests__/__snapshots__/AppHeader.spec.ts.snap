// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader.vue > matches snapshot 1`] = `
"<div data-v-48a13218="" class="header">
  <div data-v-48a13218="" class="container">
    <icon-circle-pos-stub data-v-48a13218="" class="logo"></icon-circle-pos-stub>
  </div>
</div>"
`;
